File: LeetCode/Medium/29-divide-two-integers/try-1.java
/*
 * Problem #29: Divide Two Integers
 * Difficulty: Medium
 * Submission: Try 1
 * status: Accepted
 * Language: java
 * Date: 7/28/2026, 7:27:18 PM
 * Link: https://leetcode.com/problems/divide-two-integers/
 */

class Solution {
    public int divide(int dividend, int divisor) {

        // Handle overflow case
        if (dividend == Integer.MIN_VALUE && divisor == -1)
            return Integer.MAX_VALUE;

        // Determine sign
        boolean negative = (dividend < 0) ^ (divisor < 0);

        // Convert to long to avoid overflow
        long dvd = Math.abs((long) dividend);
        long dvs = Math.abs((long) divisor);

        int ans = 0;

        while (dvd >= dvs) {
            long temp = dvs;
            int multiple = 1;

            while (dvd >= (temp << 1)) {
                temp <<= 1;
                multiple <<= 1;
            }

            dvd -= temp;
            ans += multiple;
        }

        return negative ? -ans : ans;
    }
}

File: Codeforces/263/A/try-1.py
/*
 * Problem : Beautiful Matrix
 * Difficulty: Easy
 * Submission: Try 1
 * status: Accepted
 * Language: Python 3
 * Date: 9/5/2026, 7:47:48 PM
 * Link: https://codeforces.com/contest/263/problem/A
 */

row = col = 0

for i in range(1, 6):
    nums = list(map(int, input().split()))

    for j in range(1, 6):
        if nums[j - 1] == 1:
            row = i
            col = j

moves = abs(row - 3) + abs(col - 3)

print(moves)
